apps/meteor/server/services/apps-engine/service.ts

```diff
@@ -90,6 +90,18 @@ export class AppsEngineService extends ServiceClassInternal implements IAppsEngi
 
 			// avoid updating the setting if the value is the same,
 			// which caused an infinite loop
+			// and sometimes the settings can be an array
+			// so we need to convert it to JSON stringified to compare it
+
+			if (Array.isArray(oldSetting) && Array.isArray(setting.value)) {
+				if (JSON.stringify(oldSetting) === JSON.stringify(setting.value)) {
```

Comment:
This could fail if the settings have different order, but not sure if that's really a problem.

Reply:
The order does not matter in this case, we just need to avoid it if it is strictly equal.

Reply:
No no, yeah. I was wondering because `['a','b','c']` stringified will be different from `['b','c','a']`.

Same array, different order, will say they're "different" when they're not

Reply:
I don't think this would be considered an issue here, since the settings are generated as a form to the user to interact with and changes of order really do mean that the value has changed.

Reply:
Noice, I'll resolve this then :)